SkeletalRepresentationInitializer/Logic/vtkSlicerSkeletalRepresentationInitializerLogic.cxx

```diff
@@ -1271,28 +1406,445 @@ void vtkSlicerSkeletalRepresentationInitializerLogic::AddPointToScene(double x,
+        sprintf(crestFileName, "%s/model/crest%04d.vtk", tempFolder, stepNum);
+        // read source srep
+        vtkSmartPointer<vtkPolyDataReader> upSpokeReader = vtkSmartPointer<vtkPolyDataReader>::New();
+        upSpokeReader->SetFileName(upFileName);
+        upSpokeReader->Update();
+        vtkSmartPointer<vtkPolyData> upSpokes = upSpokeReader->GetOutput();
+
+        vtkSmartPointer<vtkPolyDataReader> downSpokeReader = vtkSmartPointer<vtkPolyDataReader>::New();
+        downSpokeReader->SetFileName(downFileName);
+        downSpokeReader->Update();
+        vtkSmartPointer<vtkPolyData> downSpokes = downSpokeReader->GetOutput();
+
+        vtkSmartPointer<vtkPolyDataReader> crestSpokeReader = vtkSmartPointer<vtkPolyDataReader>::New();
+        crestSpokeReader->SetFileName(crestFileName);
+        crestSpokeReader->Update();
+        vtkSmartPointer<vtkPolyData> crestSpokes = crestSpokeReader->GetOutput();
+
+        sprintf(upFileName, "%s/model/up%04d.vtk", tempFolder, stepNum-1);
+        std::string upOutputFile(upFileName);
+        TransformNOutput(tps, upSpokes, upOutputFile);
+        // display upspokes
+
+        sprintf(downFileName, "%s/model/down%04d.vtk", tempFolder, stepNum-1);
+        std::string downOutputFile(downFileName);
+        TransformNOutput(tps, downSpokes, downOutputFile);
+
+        sprintf(crestFileName, "%s/model/crest%04d.vtk", tempFolder, stepNum-1);
+        std::string crestOutputFile(crestFileName);
+        TransformNOutput(tps, crestSpokes, crestOutputFile);
+
+        // Apply transformation matrix on the skeletal sheet and fold curve
+        char meshFileName[MAX_FILE_NAME];
+        sprintf(meshFileName, "%s/model/mesh%04d.vtk", tempFolder, stepNum); // Transform input
+        vtkSmartPointer<vtkPolyDataReader> meshReader = vtkSmartPointer<vtkPolyDataReader>::New();
+        meshReader->SetFileName(meshFileName);
+        meshReader->Update();
+        vtkSmartPointer<vtkPolyData> meshPoly = meshReader->GetOutput();
+        sprintf(meshFileName, "%s/model/mesh%04d.vtk", tempFolder, stepNum-1); // output
+        TransformPoints(tps, meshPoly, meshFileName);
+
+        char curveFileName[MAX_FILE_NAME];
+        sprintf(curveFileName, "%s/model/curve%04d.vtk", tempFolder, stepNum); // Transform input
+        vtkSmartPointer<vtkPolyDataReader> curveReader = vtkSmartPointer<vtkPolyDataReader>::New();
+        curveReader->SetFileName(curveFileName);
+        curveReader->Update();
+        vtkSmartPointer<vtkPolyData> curvePoly = curveReader->GetOutput();
+        sprintf(curveFileName, "%s/model/curve%04d.vtk", tempFolder, stepNum-1); // output
+        TransformPoints(tps, curvePoly, curveFileName);
+    }
+}
+
+int vtkSlicerSkeletalRepresentationInitializerLogic::BackwardFlow(int totalNum)
 {
+
     // 1. compute pairwise TPS
+    ComputePairwiseTps(totalNum);
+    std::cout << "Finished computing transformation matrix." << std::endl;
```

Comment:
Are these `std::cout` print expected? Or they are there just for debugging purposes?

Reply:
These are expected. Since the process of backward flow is a bit long, the printout can help to know what is going on in the backend (is the program dead or ongoing?).